opentelemetry-api/src/opentelemetry/propagators/textmap.py

```diff
@@ -92,9 +126,9 @@ class TextMapPropagator(abc.ABC):
     @abc.abstractmethod
     def extract(
         self,
-        getter: Getter[TextMapPropagatorT],
         carrier: TextMapPropagatorT,
         context: typing.Optional[Context] = None,
+        getter: Getter = default_getter,
```

Comment:
The carrier and getter should match, something like:

```python
        carrier: CarrierT,
        context: typing.Optional[Context] = None,
        getter: Getter[CarrierT] = default_getter,
```

Otherwise the TypeVars to `getter.get()` and `getter.keys()` just act as `Any`

Reply:
@aabmass, I did not have much luck incorporating these suggestions, this is what I did: https://github.com/ocelotl/opentelemetry-python/tree/aabmass_issue1644

Reply:
Also took a look and struggled to get it working. Using overloads made the call signature work well but then actually implementing a propagator became a mess (which is the part that should be easiest)

I am fine with whatever is here